src/wallet3/wallet.cpp

```diff
@@ -124,11 +177,14 @@ namespace wallet
       return;
 
     if (blocks.size() == 0)
-      //TODO: error handling; this shouldn't be able to happen
+    {
+      oxen::log::warning(logcat, "blocks size is zero, this shouldn't be able to happen");
       return;
+    }
```

Comment:
I feel like this should either be an assert (if there should be invariants in the code that make it unable to happen) or a throw (if it can happen e.g. because of bad external input).

Reply:
(As written it isn't clear which "this shouldn't be able to happen" is actually the reason it shouldn't be able to happen.)